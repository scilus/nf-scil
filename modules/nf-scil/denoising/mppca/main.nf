
process DENOISING_MPPCA {
    tag "$meta.id"
    label 'process_medium'

    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://scil.usherbrooke.ca/containers/scilus_2.0.0.sif':
        'scilus/scilus:2.0.0' }"

    input:
    tuple val(meta), path(dwi)

    output:
    tuple val(meta), path("*_dwi_denoised.nii.gz")  , emit: image
    path "versions.yml"                             , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def prefix = task.ext.prefix ?: "${meta.id}"
    def extent = task.ext.extent ? "-extent " + task.ext.extent : ""

    """
    export ITK_GLOBAL_DEFAULT_NUMBER_OF_THREADS=1
    export OMP_NUM_THREADS=1
    export OPENBLAS_NUM_THREADS=1

    dwidenoise $dwi ${prefix}_dwi_denoised.nii.gz $extent -nthreads $task.cpus
    fslmaths ${prefix}_dwi_denoised.nii.gz -thr 0 ${prefix}_dwi_denoised.nii.gz

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        mrtrix: \$(mrcalc -version 2>&1 | sed -n 's/== mrcalc \\([0-9.]\\+\\).*/\\1/p')
        fsl: \$(flirt -version 2>&1 | sed -n 's/FLIRT version \\([0-9.]\\+\\)/\\1/p')
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"

    """
    dwidenoise -h
    fslmaths -h

    touch ${prefix}_dwi_denoised.nii.gz

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        mrtrix: \$(mrcalc -version 2>&1 | sed -n 's/== mrcalc \\([0-9.]\\+\\).*/\\1/p')
        fsl: \$(flirt -version 2>&1 | sed -n 's/FLIRT version \\([0-9.]\\+\\)/\\1/p')
    END_VERSIONS
    """
}
