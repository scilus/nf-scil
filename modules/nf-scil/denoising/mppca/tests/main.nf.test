nextflow_process {

    name "Test Process DENOISING_MPPCA"
    script "../main.nf"
    process "DENOISING_MPPCA"
    config "./nextflow.config"

    tag "modules"
    tag "modules_nfcore"
    tag "denoising"
    tag "denoising/mppca"

    tag "subworkflows"
    tag "subworkflows/load_test_data"

    test("denoising - mppca") {

        setup {
            run("LOAD_TEST_DATA", alias: "LOAD_DATA") {
                script "../../../../../subworkflows/nf-scil/load_test_data/main.nf"
                process {
                    """
                    input[0] = Channel.from( [ "raw_DWIss300-dir8.zip" ] )
                    input[1] = "test.load-test-data"
                    """
                }
            }
        }

        when {
            process {
                """
                input[0] = LOAD_DATA.out.test_data_directory.map{
                    test_data_directory -> [
                        [ id:'test', single_end:false ], // meta map
                        file("\${test_data_directory}/dwi.nii.gz")
                    ]
                }
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
