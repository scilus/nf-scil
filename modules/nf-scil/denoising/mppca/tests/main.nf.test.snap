{
    "denoising - mppca": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_dwi_denoised.nii.gz:md5,f0efe6163677a0e5aca9756125f55c16"
                    ]
                ],
                "1": [
                    "versions.yml:md5,fb589ae27eb3edb1cd1a50b0e13d2714"
                ],
                "image": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_dwi_denoised.nii.gz:md5,f0efe6163677a0e5aca9756125f55c16"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,fb589ae27eb3edb1cd1a50b0e13d2714"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0-rc1",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-07-27T01:45:10.310484"
    }
}